instrumentation/src/main/java/io/opentelemetry/android/internal/features/persistence/DiskManager.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.features.persistence;
+
+import io.opentelemetry.android.config.OtelRumConfig;
+import io.opentelemetry.android.config.PersistenceConfiguration;
+import io.opentelemetry.android.internal.services.AppInfoService;
+import io.opentelemetry.android.internal.services.PreferencesService;
+import io.opentelemetry.android.internal.services.Service;
+import io.opentelemetry.android.internal.services.ServiceManager;
+import java.io.File;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class is internal and not for public use. Its APIs are unstable and can change at any time.
+ */
+public final class DiskManager {
+    private final AppInfoService appInfoService;
+    private final PreferencesService preferencesService;
+    private final PersistenceConfiguration persistenceConfiguration;
+    private static final String MAX_FOLDER_SIZE_KEY = "max_signal_folder_size";
+    private static final int MAX_FILE_SIZE = 1024 * 1024;
+    private final Logger logger = Logger.getLogger("DiskManager");
+
+    public static DiskManager create(OtelRumConfig config) {
+        ServiceManager serviceManager = ServiceManager.get();
+        return new DiskManager(
+                serviceManager.getService(Service.Type.APPLICATION_INFO),
+                serviceManager.getService(Service.Type.PREFERENCES),
+                config.getPersistenceConfiguration());
+    }
+
+    DiskManager(
```

Comment:
Yeah, I just double-checked why is it like that and I remembered is because I needed it visible for tests (this code comes from the Elastic repo) so, depending on the test strategy we want to go with here, it might not be needed to have it package-private. Which brings me to the fact that I should update this PR to include tests since the overall approach was well received.

Reply:
It's done now 👍 - I've added tests while also making the constructor private.